zenoh.json5

```diff
@@ -0,0 +1,79 @@
+    storage_manager: {
+      volumes: {
+        s3: {
+
+            // AWS region to which connect (see https://docs.aws.amazon.com/general/latest/gr/s3.html). 
+            // This field is mandatory if you are going to communicate with an AWS S3 server and
+            // optional in case you are working with a MinIO S3 server.
+            region: "eu-west-1",
+
+            // Endpoint where the S3 server is located.
+            // This parameter allows you to specify a custom endpoint when working with a MinIO S3 
+            // server.
+            // This field is mandatory if you are working with a MinIO server and optional in case
+            // you are working with an AWS S3 server as long as you specified the region, in which
+            // case the endpoint will be resolved automatically.
+            url: "https://s3.eu-west-1.amazonaws.com",
+
+            private: {
+                // Credentials for interacting with the S3 volume. They may differ from the storage
+                // credentials.
+                access_key: "AKIAIOSFODNN7EXAMPLE",
+                secret_key: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+            }
+        }
+      },
+      storages: {
+        // Configuration of a "demo" storage using the S3 volume. Each storage is associated to a
+        // single S3 bucket.
+        s3_storage: {
+
+          // The key expression this storage will subscribes to
+          key_expr: "s3/example/*",
+
+          // this prefix will be stripped from the received key when converting to database key.
+          // i.e.: "demo/example/a/b" will be stored as "a/b"
+          strip_prefix: "s3/example",
+
+          volume: {
+            // Id of the volume this storage is associated to
+            id: "s3",
+
+            // Bucket to which this storage is associated to
+            bucket: "zenoh-bucket",
+
+            // The storage attempts to create the bucket, but if the bucket already exists and is
+            // owned by you, then with 'reuse_bucket' you can associate that preexisting bucket to
+            // the storage, otherwise it will fail.
+            reuse_bucket: true,
+
+            // If the storage is read only, it will only handle GET requests
+            read_only: false,
+
+            // strategy on storage closure, either `destroy_bucket` or `do_nothing`
+            // adminspace.permissions.write needs to be set to true to destroy the bucket on closure
+            on_closure: "destroy_bucket",
+
+            private: {
+                // Credentials for interacting with the S3 bucket
+                access_key: "AKIAIOSFODNN7EXAMPLE",
+                secret_key: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+            }
+          }
+        },
+      }
+    },
+    // Optionally, add the REST plugin
+    rest: { http_port: 8000 }
+  },
+  adminspace: {
+    // read and/or write permissions on the admin space
+    permissions: {
+      read: true,
+      write: true,
```

Comment:
Is there a specific reason for needing `write` access to the adminspace?

Reply:
Personally I don't like the idea of granting `write` permission on the whole admin space because of a plugin.
Do you think is it possible to relax this requirement?

Reply:
I don't think it's possible by modifying the plugin codebase. If I'm not wrong, we need to modify something inside zenoh in order to relax this, probably in the zenoh-plugin-storage-manager.

Reply:
@JEnoch any comments on this?

Reply:
The `write` permission on admin space only establishes if volume/storage creation and deletion is allowed at runtime.
If `write == false`, then the volumes/storages are created at `zenohd` startup and deleted when it stops.
Thus whether write permission is granted or not should have no impact on the fact that at some point the storage is deleted and some clean-up might be required at this point (depending the `on_closure` configuration).

What is probably missing now is a proper handling of stop signals in `zenohd` that would trigger a proper halt of the `storage-manager` plugin and proper deletion of volumes/storages. But that's independent of the admin space permissions.